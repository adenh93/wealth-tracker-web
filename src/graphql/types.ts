export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type CryptoCurrency = {
  __typename?: 'CryptoCurrency';
  id: Scalars['Int'];
  name: Scalars['String'];
  rank: Scalars['Int'];
  symbol: Scalars['String'];
};

export type CryptoCurrencyHolding = {
  __typename?: 'CryptoCurrencyHolding';
  cryptoCurrency?: Maybe<CryptoCurrency>;
  cryptoCurrencyId: Scalars['Int'];
  holdings: Scalars['Float'];
  id: Scalars['Int'];
  netHoldingsValue: Scalars['Float'];
  percentChange24h: Scalars['Float'];
  price: Scalars['Float'];
};

export type CryptoCurrencyHoldingsAddInput = {
  cryptoCurrencyId: Scalars['Int'];
  holdings: Scalars['Float'];
};

export type CryptoCurrencyHoldingsUpdateInput = {
  holdings: Scalars['Float'];
  id: Scalars['Int'];
};

export type Mutation = {
  __typename?: 'Mutation';
  cryptoCurrencyHoldingsAdd?: Maybe<CryptoCurrencyHolding>;
  cryptoCurrencyHoldingsDelete?: Maybe<CryptoCurrencyHolding>;
  cryptoCurrencyHoldingsUpdate?: Maybe<CryptoCurrencyHolding>;
};


export type MutationCryptoCurrencyHoldingsAddArgs = {
  input: CryptoCurrencyHoldingsAddInput;
};


export type MutationCryptoCurrencyHoldingsDeleteArgs = {
  id: Scalars['Int'];
};


export type MutationCryptoCurrencyHoldingsUpdateArgs = {
  input: CryptoCurrencyHoldingsUpdateInput;
};

export type Query = {
  __typename?: 'Query';
  cryptoCurrencies?: Maybe<Array<CryptoCurrency>>;
  cryptoCurrencyHoldings?: Maybe<Array<CryptoCurrencyHolding>>;
};


export type QueryCryptoCurrenciesArgs = {
  query?: Maybe<Scalars['String']>;
};
