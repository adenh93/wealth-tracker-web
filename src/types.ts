export interface Asset {
  id: string
  logoSrc: string
  name: string
  ticker: string
  price: number
  gain24Hr: number
  holdings: number
}
