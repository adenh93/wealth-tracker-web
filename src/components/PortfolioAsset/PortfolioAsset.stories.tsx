import PortfolioAsset, { PortfolioAssetProps } from './PortfolioAsset'
import { Story } from '@storybook/react'
import { Asset } from '../../types'

const config = {
  title: 'Components/PortfolioAsset',
  component: PortfolioAsset,
}

const Template: Story<PortfolioAssetProps> = (args) => (
  <PortfolioAsset {...args} />
)

export const Primary = Template.bind({})

const asset: Asset = {
  logoSrc: 'https://upload.wikimedia.org/wikipedia/commons/4/4e/BitConnect.png',
  name: 'Bitconnect',
  ticker: 'BCC',
  price: 0.003021,
  gain24Hr: -0.9984,
  holdings: 2450000.15,
}

Primary.args = {
  asset,
}

export default config
