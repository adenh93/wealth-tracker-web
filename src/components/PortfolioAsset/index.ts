export { default } from './PortfolioAsset'
export type { PortfolioAssetProps } from './PortfolioAsset'