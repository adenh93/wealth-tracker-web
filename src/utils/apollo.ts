import { ApolloError } from '@apollo/client'

export const extractInputErrorField = <FormType>(
  error: ApolloError
): keyof FormType | null => {
  if (error.graphQLErrors.length)
    return error.graphQLErrors[0].extensions?.field as keyof FormType
  return null
}
