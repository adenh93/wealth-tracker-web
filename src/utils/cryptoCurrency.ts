import { CMC_LOGO_URL } from '../constants'

export const getCMCLogoUrl = (id: number): string => `${CMC_LOGO_URL}/${id}.png`
