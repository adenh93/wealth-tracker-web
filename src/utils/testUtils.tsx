import { ReactElement } from 'react'
import { ThemeProvider } from '@mui/material'
import { render as tlRender, RenderResult } from '@testing-library/react'
import theme from '../theme'

export const render = (component: ReactElement): RenderResult =>
  tlRender(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
