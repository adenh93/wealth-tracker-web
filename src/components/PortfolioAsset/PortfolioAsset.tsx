import { FC } from 'react'
import { Button, TableCell, Box } from '@mui/material'
import { OwnedAsset } from '../../types'
import { formatCurrency, formatNumber } from '../../utils/number'
import ProfitLossLabel from '../ProfitLossLabel'
import TableBodyText from '../typography/TableBodyText'
import TableBodySubText from '../typography/TableBodySubText'
import { StyledTableRow, AssetInfoWrapper } from './PortfolioAsset.css'

export interface PortfolioAssetProps {
  asset: OwnedAsset
}

const PortfolioAsset: FC<PortfolioAssetProps> = ({ asset }) => (
  <StyledTableRow>
    <TableCell align="left">
      <AssetInfoWrapper>
        <img
          width={30}
          height={30}
          src={asset.logoSrc}
          alt={`${asset.name} logo`}
        />
        <Box>
          <TableBodyText>{asset.name}</TableBodyText>
          <TableBodySubText>{asset.ticker}</TableBodySubText>
        </Box>
      </AssetInfoWrapper>
    </TableCell>
    <TableCell align="left">
      <TableBodyText>
        {formatCurrency(asset.price, { maximumFractionDigits: 8 })}
      </TableBodyText>
      <ProfitLossLabel
        value={asset.gain24Hr}
        type="percent"
        variant="tableCell2"
      />
    </TableCell>
    <TableCell align="left">
      <TableBodyText>
        {formatCurrency(asset.price * asset.holdings, {
          maximumFractionDigits: 2,
        })}
      </TableBodyText>
      <TableBodySubText>{formatNumber(asset.holdings)}</TableBodySubText>
    </TableCell>
    <TableCell align="left">
      <Button size="small" variant="contained" sx={{ marginRight: 1 }}>
        Edit
      </Button>
      <Button size="small" variant="contained" color="secondary">
        Remove
      </Button>
    </TableCell>
  </StyledTableRow>
)

export default PortfolioAsset
