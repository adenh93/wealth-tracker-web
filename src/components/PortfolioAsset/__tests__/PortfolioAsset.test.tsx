import { render } from '../../../utils/testUtils'
import { Asset } from '../../../types'
import PortfolioAsset from '..'

describe('Portfolio Asset Tests', () => {
  const mockAsset: Asset = {
    logoSrc: 'https://localhost:3000/image.png',
    name: 'Apple Inc.',
    ticker: 'AAPL',
    price: 134.59,
    gain24Hr: 0.014,
    holdings: 42000,
  }

  test('matches snapshot', () => {
    const { asFragment } = render(<PortfolioAsset asset={mockAsset} />)

    expect(asFragment).toMatchSnapshot()
  })

  test('renders properly', () => {
    const { getByText } = render(<PortfolioAsset asset={mockAsset} />)

    expect(getByText('Apple Inc.')).not.toBeNull()
    expect(getByText('AAPL')).not.toBeNull()
    expect(getByText('A$134.59')).not.toBeNull()
    expect(getByText('+ 1.4%')).not.toBeNull()
    expect(getByText('A$5,652,780.00')).not.toBeNull()
    expect(getByText('42,000')).not.toBeNull()
  })
})
