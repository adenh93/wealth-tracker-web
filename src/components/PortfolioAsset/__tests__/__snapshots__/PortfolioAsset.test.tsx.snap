// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio Asset Tests matches snapshot 1`] = `[Function]`;
